public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -642,33 +645,42 @@ export class LokiDatasource
    * Implemented as part of DataSourceAPI and used for template variable queries.
    * @returns A Promise that resolves to an array of results from the metric find query.
    */
-  async metricFindQuery(query: LokiVariableQuery | string, options?: LegacyMetricFindQueryOptions) {
+  async metricFindQuery(
+    query: LokiVariableQuery | string,
+    options?: LegacyMetricFindQueryOptions
+  ): Promise<MetricFindValue[]> {
     if (!query) {
       return Promise.resolve([]);
     }
 
+    let interpolatedVariableQuery: LokiVariableQuery | undefined;
+
     if (typeof query === 'string') {
-      const interpolated = this.interpolateString(query, options?.scopedVars);
-      return await this.legacyProcessMetricFindQuery(interpolated);
+      interpolatedVariableQuery = this.parseStringToVariableQuery(this.interpolateString(query, options?.scopedVars));
```

Comment:
Great change! Makes it easier to understand what is doing without looking at the function.